packages/lib/test/ecdsa.test.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  privateKeyToPublicKey,
+  recoverPubKeyIndexFromSignature,
+  verifyEcdsaSignature,
+} from "../src/ecdsa";
+import { hexToBigInt } from "../src/utils";
+
+describe("javascript signature verification", () => {
+  test("should verify a baby jubjub ecdsa signature 0", () => {
+    const privKey =
+      "0323dbbda9a5aff570d974d71c88334cf99ab9c0455e1d2546ca03ca069eb1e0";
+    const msgHash = BigInt("0");
+    const sig = {
+      r: hexToBigInt(
+        "00EF7145470CEC0B683C629CBA8ED58110000FFE657366F7D5A91F2D149DD8B5"
+      ),
+      s: hexToBigInt(
+        "0370C60A23266F520C56DA088B4C4AFAAAF6BB1993A501980F6D8FB6F343984A"
+      ),
+    };
+
+    const pubKey = privateKeyToPublicKey(hexToBigInt(privKey));
+
+    expect(verifyEcdsaSignature(sig, msgHash, pubKey)).toBe(true);
+  });
+
+  test("should verify a baby jubjub ecdsa signature 1", () => {
+    const privKey =
+      "0323dbbda9a5aff570d974d71c88334cf99ab9c0455e1d2546ca03ca069eb1e0";
+    const msgHash = BigInt("1");
+    const sig = {
+      r: hexToBigInt(
+        "04BEF5B82A7637BBFF0D3C52DDB982A00C84FE8A386625369B511CF538CD3584"
+      ),
+      s: hexToBigInt(
+        "00CA8ED01E70CEC6DE27C1B9F6735B52FB49E4521F50BEEDEED8E81459729E2E"
+      ),
+    };
+
+    const pubKey = privateKeyToPublicKey(hexToBigInt(privKey));
+
+    expect(verifyEcdsaSignature(sig, msgHash, pubKey)).toBe(true);
+  });
+
+  test("should verify a baby jubjub ecdsa signature 2", () => {
```

Comment:
Can we also add a test using some of the abadbabe signatures?

Reply:
Added